lib/api/response/base/api_response.dart

```diff
@@ -18,10 +19,18 @@ class ApiResponse<T extends BaseResponse> {
   Map<String, dynamic> toJson() => _$ApiResponseToJson(this);
 
   static T _dataFromJson<T>(Map<String, dynamic> json) {
-    throw Exception("_dataFromJson Not supported response type");
+    if (T == OauthTokenResponse) {
+      return OauthTokenResponse.fromJson(json) as T;
+    } else {
+      throw Exception("_dataFromJson Not supported response type");
```

Comment:
I think we can create a constant for this exception, we're repeating this duplicate code. What do you think? 🤔

Reply:
@markgravity repeating? 🤔 I have 2 places of it: 1 for _dataFromJson, 1 for _dataToJson with different messages. I'm not sure should we still try to create a const for them 💭

Reply:
@luongvo oh you're right. I gotta refresh my eyes 🤣